### I- Scores avant ajout du SQL sur le RAG

#### 1 - Scores détaillés par ligne 

In [13]:
import pandas as pd

In [14]:
score_1 = pd.read_csv("ragas_results.csv")
score_1

,question,answer,contexts,ground_truth,faithfulness,answer_relevancy,context_precision,context_recall
0,Quel joueur a le meilleur pourcentage à 3 poin...,"D'après les données disponibles, Shai Gilgeous...",['Analyse des performances des joueurs en tena...,Alondes Williams (DET) affiche un pourcentage ...,0.5,0.709660,1.0,0.000000
1,Quel joueur a le meilleur % de réussite aux la...,"D'après les données disponibles, Shai Gilgeous...",['Comparaison des performances des joueurs en ...,Sam Hauser (BOS) affiche un taux de réussite d...,1.0,0.901402,1.0,0.333333
2,Quel joueur a le plus de points combinés (poin...,"D'après les données disponibles, Nikola Jokić ...",['Agrégation de plusieurs statistiques pour év...,Nikola Jokić domine nettement avec 3675 unités...,NaN,0.000000,1.0,0.333333
3,Quel joueur a la meilleure efficacité offensiv...,"D'après les données disponibles, Joel Embiid (...",['Analyse des performances ciblées contre des ...,Giannis Antetokounmpo (MIL) est le joueur le p...,1.0,0.754911,1.0,0.000000
4,Quel joueur a amélioré son pourcentage à 2 poi...,"D'après les données disponibles, Shai Gilgeous...","[""Analyse de l'évolution des statistiques d'un...",Alondes Williams (DET) présente un taux de réu...,1.0,0.000000,0.0,0.000000
5,Quelle est la durée légale d’un CDD en France ?,"En tant qu'analyste NBA, je ne suis pas compét...",['La durée maximale d’un CDD en France est gén...,La durée maximale d’un CDD est en général de 1...,NaN,0.000000,1.0,1.000000
6,Comment obtenir un passeport biométrique ?,"En tant qu'analyste NBA, je me concentre sur l...","[""Un passeport biométrique s'obtient en mairie...",L’obtention d’un passeport biométrique nécessi...,1.0,0.860621,1.0,1.000000


#### 2- Scores globaux

In [15]:
score_1_1 = pd.read_csv("ragas_summary.csv")
score_1_1

,Metric,Average,Std
0,faithfulness,0.900000,0.223607
1,answer_relevancy,0.460942,0.435781
2,context_precision,0.857143,0.377964
3,context_recall,0.380952,0.448395


### II- Scores avant ajout du SQL sur le RAG

#### 1- Scores détaillés

In [16]:
score_2 = pd.read_csv("ragas_results_2.csv")
score_2

,question,answer,contexts,ground_truth,faithfulness,answer_relevancy,context_precision,context_recall
0,Quel joueur a le meilleur pourcentage à 3 poin...,Alondes Williams (DET) affiche un incroyable 1...,['Analyse des performances des joueurs en tena...,Alondes Williams (DET) affiche un pourcentage ...,1.0,0.765055,0.0,0.000000
1,Quel joueur a le meilleur % de réussite aux la...,Sam Hauser (BOS) affiche un taux de réussite p...,['Comparaison des performances des joueurs en ...,Sam Hauser (BOS) affiche un taux de réussite d...,1.0,0.779099,1.0,0.333333
2,Quel joueur a le plus de points combinés (poin...,Nikola Jokić domine largement avec 3675 points...,['Agrégation de plusieurs statistiques pour év...,Nikola Jokić domine nettement avec 3675 unités...,1.0,0.779061,1.0,0.333333
3,Quel joueur a la meilleure efficacité offensiv...,Giannis Antetokounmpo (MIL) est le joueur le p...,['Analyse des performances ciblées contre des ...,Giannis Antetokounmpo (MIL) est le joueur le p...,1.0,0.957428,1.0,0.500000
4,Quel joueur a amélioré son pourcentage à 2 poi...,Alondes Williams (DET) affiche un pourcentage ...,"[""Analyse de l'évolution des statistiques d'un...",Alondes Williams (DET) présente un taux de réu...,1.0,0.758162,1.0,0.000000
5,Quelle est la durée légale d’un CDD en France ?,Ce domaine ne relève pas de mes compétences. V...,['La durée maximale d’un CDD en France est gén...,La durée maximale d’un CDD est en général de 1...,NaN,0.000000,1.0,1.000000
6,Comment obtenir un passeport biométrique ?,Ce domaine ne relève pas de mes compétences. V...,"[""Un passeport biométrique s'obtient en mairie...",L’obtention d’un passeport biométrique nécessi...,NaN,0.000000,1.0,1.000000


#### 2- Scores globaux

In [18]:
score_2_1 = pd.read_csv("ragas_summary_2.csv")
score_2_1

,Metric,Average,Std
0,faithfulness,1.000000,0.000000
1,answer_relevancy,0.576972,0.400092
2,context_precision,0.857143,0.377964
3,context_recall,0.452381,0.416270


### III- Interprétations générales

✅ **Comparaison des performances RAG — Avant vs Après SQL**

1. Faithfulness (fidélité factuelle)

    - Score avant SQL	:   0.90
    - Score après SQL	:   1.00

**Analyse**

L’ajout de la base SQL fait passer la fidélité à 100%, ce qui signifie que :

les réponses du modèle sont désormais toujours factuellement alignées avec la vérité terrain (ground truth),

le système évite mieux les hallucinations,

la requête SQL apporte une source structurée et fiable, ce qui rend les réponses beaucoup plus exactes.

***C’est la plus forte amélioration observée.***

2. Answer relevancy (pertinence de la réponse)

    - Score avant SQL	0.4609
    - Score après SQL	0.5769

**Analyse**

Une amélioration d’environ +25%, ce qui montre que :

les réponses sont maintenant plus en phase avec ce que l’utilisateur demande,

le modèle comprend mieux la question car il peut s'appuyer sur des données SQL structurées,

la présence de vraies statistiques issues de la base SQL donne des réponses plus ciblées.


➡️ Le système devient plus utile, pas seulement plus exact.

3. Context Precision

    - Score avant SQL	0.8571
    - Score après SQL	0.8571

**Analyse**

Le score ne change pas.

C’est logique :

On n'a pas modifié la logique de sélection des chunks RAG initiaux,

l’ajout de SQL n’a fait qu’ajouter une autre source, mais n’a pas modifié la précision des contextes RAG envoyés au modèle.

➡️ Stabilité normale.

4. Context Recall

    - Score avant SQL	0.3809
    - Score après SQL	0.4523

**Analyse**

Une amélioration de +18%, signe que :

les réponses utilisent désormais plus de contextes pertinents,

les requêtes SQL aident à compléter les trous d’information laissés par les embeddings,

le modèle puise dans plus de sources utiles pour répondre correctement.

➡️ Le système retrouve plus d’informations pertinentes.

**Interprétation générale**

L’ajout de la base SQL a :

Éliminé les hallucinations (faithfulness = 1.00),

Rendu les réponses plus pertinentes (answer relevancy ↑),

Amélioré l’exploitation du contexte (recall ↑),

Sans changer la précision du contexte (precision stable).

Le RAG est maintenant bien meilleur, plus fiable et plus utile.